lte/gateway/c/oai/tasks/sctp/sctp_itti_messaging.c

```diff
@@ -28,39 +28,83 @@
 #include "intertask_interface.h"
 #include "sctp_itti_messaging.h"
 #include "itti_types.h"
+#include "log.h"
 #include "sctp_messages_types.h"
 
 //------------------------------------------------------------------------------
 int sctp_itti_send_lower_layer_conf(
-    task_id_t origin_task_id, sctp_assoc_id_t assoc_id, sctp_stream_id_t stream,
-    uint32_t mme_ue_s1ap_id, bool is_success) {
+    task_id_t origin_task_id, sctp_ppid_t ppid, sctp_assoc_id_t assoc_id,
+    sctp_stream_id_t stream, uint32_t ap_id, bool is_success) {
   MessageDef* msg = itti_alloc_new_message(TASK_SCTP, SCTP_DATA_CNF);
 
-  SCTP_DATA_CNF(msg).assoc_id       = assoc_id;
-  SCTP_DATA_CNF(msg).stream         = stream;
-  SCTP_DATA_CNF(msg).mme_ue_s1ap_id = mme_ue_s1ap_id;
-  SCTP_DATA_CNF(msg).is_success     = is_success;
+  SCTP_DATA_CNF(msg).ppid     = ppid;
+  SCTP_DATA_CNF(msg).assoc_id = assoc_id;
+  SCTP_DATA_CNF(msg).stream   = stream;
+
+switch(ppid) {
+    case S1AP:{	
+    OAILOG_DEBUG(
+        LOG_SCTP, "Ppid S1AP in sctp_itti_send_lower_layer_conf ");
+    SCTP_DATA_CNF(msg).mme_ue_s1ap_id = ap_id;
+    }
+    break;
+    
+    case NGAP:{	
+    OAILOG_DEBUG(
+        LOG_SCTP, " Ppid NGAP in sctp_itti_send_lower_layer_conf ");
+    SCTP_DATA_CNF(msg).amf_ue_ngap_id = ap_id;
+    }
+    break;
+    
+    default:	
+    OAILOG_ERROR(
+        LOG_SCTP, "Ppid: %d not matching in sctp_itti_send_lower_layer_conf ",
+        ppid);
+    return RETURNerror;
+  }
+
+  SCTP_DATA_CNF(msg).is_success = is_success;
```

Comment:
once you make the changes I requested, there is no need to change the body except for:
`SCTP_DATA_CNF(msg).agw_ue_xap_id = ap_id;`